modules/apps/27-interchain-accounts/host/keeper/keeper.go

```diff
@@ -199,3 +205,33 @@ func (k Keeper) SetInterchainAccountAddress(ctx sdk.Context, connectionID, portI
 	store := ctx.KVStore(k.storeKey)
 	store.Set(icatypes.KeyOwnerAccount(portID, connectionID), []byte(address))
 }
+
+// GetAuthority returns the 27-interchain-accounts host submodule's authority.
+func (k Keeper) GetAuthority() string {
+	return k.authority
+}
+
+// GetParams returns the total set of the host submodule parameters.
+func (k Keeper) GetParams(ctx sdk.Context) types.Params {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get([]byte(types.ParamsKey))
+	if bz == nil {
+		return types.Params{}
+	}
+
+	var params types.Params
+	k.cdc.MustUnmarshal(bz, &params)
+	return params
+}
+
+// SetParams sets the total set of the host submodule parameters.
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	if err := params.Validate(); err != nil {
+		return err
+	}
```

Comment:
@damiannolan do you have any thoughts on this? This pattern goes against our normal behaviour (normally all validation is expected to happen before calling `Set`). I'm fine with it but it will likely lead to redundant checks

Reply:
After thinking about this some more, I have a slight preference for maintaining our existing design pattern of doing validation before calling the set function. I'm not sure there's a benefit to doing an extra validation before setting (unless you assume there's a missing validation check in our code)

Reply:
imho I prefer to have setters with a void return arg.